apps/spruce/src/components/Spawn/utils.ts

```diff
@@ -21,3 +29,101 @@ export const getDefaultExpiration = () => {
   nextWeek.setDate(nextWeek.getDate() + 7);
   return nextWeek.toString();
 };
+
+type ValidateInput = {
+  enabledWeekdays: boolean[];
+  endTime: string;
+  runContinuously: boolean;
+  startTime: string;
+  useDefaultUptimeSchedule: boolean;
+};
+
+export const validateUptimeSchedule = ({
+  enabledWeekdays,
+  endTime,
+  runContinuously,
+  startTime,
+  useDefaultUptimeSchedule,
+}: ValidateInput): {
+  enabledHoursCount: number;
+  errors: string[];
+  warnings: string[];
+} => {
+  const { enabledHoursCount, enabledWeekdaysCount } = getEnabledHoursCount({
+    enabledWeekdays,
+    endTime,
+    runContinuously,
+    startTime,
+  });
+
+  if (useDefaultUptimeSchedule) {
+    return {
+      enabledHoursCount,
+      errors: [],
+      warnings: [],
+    };
+  }
```

Comment:
It's actually possible that the user would manually edit the sleep schedule to, then decide to use the default schedule. The form would still include the manual values so if they were invalid it would register as an error even though they're opting for the default.